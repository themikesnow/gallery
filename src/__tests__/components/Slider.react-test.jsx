import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { expect as expectChai } from 'chai';
import Slider from '../../components/Gallery/Slider.react';

const image = {
  src: 'DummyPath',
};

describe('<Slider />', () => {
  it('Match Snapshot', () => {
    const onPreviousMock = jest.fn();
    const onNextMock = jest.fn();
    let component = renderer.create(<Slider image={image} onPreviousImage={onPreviousMock} onNextImage={onNextMock} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<Slider image={image} onPreviousImage={onPreviousMock} onNextImage={onNextMock} />);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls props onPreviousImage', () => {
    const onPreviousMock = jest.fn();
    const onNextMock = jest.fn();
    const wrapper = mount(<Slider image={image} onPreviousImage={onPreviousMock} onNextImage={onNextMock} />);
    wrapper.find('.btn-previous').simulate('click');

    expectChai(onPreviousMock.mock.calls.length).to.equal(1);
  });

  it('calls props onNextImage', () => {
    const onPreviousMock = jest.fn();
    const onNextMock = jest.fn();
    const wrapper = mount(<Slider image={image} onPreviousImage={onPreviousMock} onNextImage={onNextMock} />);
    wrapper.find('.btn-next').simulate('click');

    expectChai(onNextMock.mock.calls.length).to.equal(1);
  });


  it('Updates loading flag when new image is received', () => {
    const onPreviousMock = jest.fn();
    const onNextMock = jest.fn();
    const wrapper = mount(<Slider image={image} onPreviousImage={onPreviousMock} onNextImage={onNextMock} />);
    wrapper.setProps({ image: { src: 'Another DummyPath' } });
    expectChai(wrapper.state().isLoading).to.equal(true);
  });
});
