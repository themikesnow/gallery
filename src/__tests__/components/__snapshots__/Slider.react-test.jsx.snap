// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Match Snapshot 1`] = `
<div
  className="slider"
>
  <button
    className="btn btn-previous disabled"
    onClick={[Function]}
  >
    Previuos
  </button>
  <button
    className="btn btn-next disabled"
    onClick={[Function]}
  >
    Next
  </button>
  <div
    className="loading"
  >
    <i
      className="fa fa-refresh fa-spin fa-3x fa-fw"
    />
  </div>
  <img
    alt="Slider"
    className="hidden"
    onLoad={[Function]}
    src="DummyPath"
  />
</div>
`;

exports[`<Slider /> Match Snapshot 2`] = `
<div
  className="slider"
>
  <button
    className="btn btn-previous disabled"
    onClick={[Function]}
  >
    Previuos
  </button>
  <button
    className="btn btn-next disabled"
    onClick={[Function]}
  >
    Next
  </button>
  <div
    className="loading"
  >
    <i
      className="fa fa-refresh fa-spin fa-3x fa-fw"
    />
  </div>
  <img
    alt="Slider"
    className="hidden"
    onLoad={[Function]}
    src="DummyPath"
  />
</div>
`;
